prime-router/src/main/kotlin/azure/observability/event/ReportStreamEventData.kt

```diff
@@ -0,0 +1,118 @@
+package gov.cdc.prime.router.azure.observability.event
+
+import com.fasterxml.jackson.annotation.JsonKey
+import com.fasterxml.jackson.annotation.JsonUnwrapped
+import com.google.common.base.CaseFormat
+import gov.cdc.prime.router.Topic
+import gov.cdc.prime.router.azure.db.enums.TaskAction
+import java.time.OffsetDateTime
+import java.util.UUID
+
+/**
+ * Data class that is used in all ReportStream report events
+ *
+ * @param childReportId the report id of the outputted report
+ * @param parentReportId the optional report if of the inputted report
+ * @param submittedReportIds all the submitted reports that the outputted report ids has items from.
+ * @param topic the [Topic] that the report is part of
+ * @param blobUrl the blob url for the outputted report
+ * @param pipelineStepName the step that produced the outputted report
+ * @param timestamp when the event occurred
+ */
+data class ReportEventData(
+    val childReportId: UUID,
+    val parentReportId: UUID?,
+    val submittedReportIds: List<UUID>,
+    val topic: Topic?,
+    val blobUrl: String,
+    val pipelineStepName: TaskAction,
+    val timestamp: OffsetDateTime,
+)
+
+/**
+ * Data class is used in all ReportStream item events
+ *
+ * @param childItemIndex the index of the outputted item in the child report
+ * @param parentItemIndex the index the item in the input report
+ * @param submittedItemIndex the index of the item in the submitted report
+ * @param trackingId a unique identifier for the item as it goes through the pipeline
+ * @param sender the sender of the item
+ */
+data class ItemEventData(
+    val childItemIndex: Int,
+    val parentItemIndex: Int?,
+    val submittedItemIndex: Int?,
+    val trackingId: String?,
+    val sender: String,
+)
+
+/**
+ * This enum contains properties values that can be used in creating params for ReportStream events
+ * @see [AbstractReportStreamEventBuilder.params]
+ */
+enum class ReportStreamEventProperties {
+    PROCESSING_ERROR,
+    ITEM_FORMAT,
+    VALIDATION_PROFILE,
+    FAILING_FILTERS,
+    FILTER_TYPE,
+    FILENAME,
+    TRANSPORT_TYPE,
+    RECEIVER_NAME,
+    REQUEST_PARAMETERS,
+    SENDER_IP,
+    FILE_LENGTH,
+    SENDER_NAME,
+    BUNDLE_DIGEST,
+    ;
+
+    @JsonKey
+    fun externalKey(): String {
```

Comment:
This annotation means that when the enum is used as a key we'll go from `PROCESSING_ERROR` -> `processingError`

Reply:
Didn't know you could do this, awesome!